server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorBalancer.java

```diff
@@ -128,16 +128,25 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
 
           if (holder != null) {
             moveSegment(segmentToMove, holder.getServer(), params);
+          } else {
+            ++unmoved;
           }
         }
       }
+      if(unmoved == maxSegmentsToMove) {
+        // Cluster should be alive and constantly adjusting
+        log.warn("No good moves found in tier [%s]", tier);
```

Comment:
Info sounds ok to me, that resolves my concern that people would be spooked by warnings about conditions that are not necessarily bad.